src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1515 @@
+        hs = self.sim.modules["HealthSystem"]
+        p = self.sim.modules['CervicalCancer'].parameters
+
+        # todo: request consumables needed for this
+
+        if not df.at[person_id, 'is_alive']:
+            return hs.get_blank_appt_footprint()
+
+        # If the status is already in `stage4`, start palliative care (instead of treatment)
+        if df.at[person_id, "ce_hpv_cc_status"] == 'stage4':
+            logger.warning(key="warning", data="Cancer is in stage 4 - aborting HSI_CervicalCancer_StartTreatment,"
+                                               "scheduling HSI_CervicalCancer_PalliativeCare")
+
+            hs.schedule_hsi_event(
+                hsi_event=HSI_CervicalCancer_PalliativeCare(
+                     module=self.module,
+                     person_id=person_id,
+                ),
+                topen=self.sim.date,
+                tclose=None,
+                priority=0
+            )
+            return self.make_appt_footprint({})
+
+        # Check that the person has been diagnosed and is not on treatment
+        assert not pd.isnull(df.at[person_id, "ce_date_diagnosis"])
+
+        # Record date and stage of starting treatment
+        df.at[person_id, "ce_date_treatment"] = self.sim.date
+        df.at[person_id, "ce_ever_treated"] = True
+        df.at[person_id, "ce_stage_at_which_treatment_given"] = df.at[person_id, "ce_hpv_cc_status"]
+
+        # stop vaginal bleeding
+        self.sim.modules['SymptomManager'].change_symptom(
+            person_id=person_id,
+            symptom_string='vaginal_bleeding',
+            add_or_remove='-',
+            disease_module=self.module
+            )
+
+        random_value = random.random()
+
+        if random_value <= p['prob_cure_stage1'] and df.at[person_id, "ce_date_treatment"] == self.sim.date:
+            df.at[person_id, "ce_hpv_cc_status"] = 'none'
+            df.at[person_id, 'ce_current_cc_diagnosed'] = False
+        else:
+            df.at[person_id, "ce_hpv_cc_status"] = 'stage1'
+
+        if random_value <= p['prob_cure_stage2a'] and df.at[person_id, "ce_date_treatment"] == self.sim.date:
+            df.at[person_id, "ce_hpv_cc_status"] = 'none'
+            df.at[person_id, 'ce_current_cc_diagnosed'] = False
+        else:
+            df.at[person_id, "ce_hpv_cc_status"] = 'stage2a'
+
+        if random_value <= p['prob_cure_stage2b'] and df.at[person_id, "ce_date_treatment"] == self.sim.date:
+            df.at[person_id, "ce_hpv_cc_status"] = 'none'
+            df.at[person_id, 'ce_current_cc_diagnosed'] = False
+        else:
+            df.at[person_id, "ce_hpv_cc_status"] = 'stage2b'
+
+        if random_value <= p['prob_cure_stage3'] and df.at[person_id, "ce_date_treatment"] == self.sim.date:
+            df.at[person_id, "ce_hpv_cc_status"] = 'none'
+            df.at[person_id, 'ce_current_cc_diagnosed'] = False
+        else:
+            df.at[person_id, "ce_hpv_cc_status"] = 'stage3'
```

Comment:
Are we assuming that patients do recover from cervical cancer same day they receive treatment?

Reply:
don;t quite understand this passage. should we not be checking what stage the person is in? I presume the 'prob_cure_stageX' correspond only to persons in stage X?

Reply:
Ah yes - thanks for spotting that